src/components/HeroCopyElement/index.js

```diff
@@ -1,76 +1,207 @@
+    delay: 1,
+  };
+  const transitionTitleMobile = {
+    yoyo: 1,
+    duration: 0.5,
+    ease: "easeInOut",
+    repeatDelay: 1.75,
+    delay: 1,
+  };
+  const transitionLine = {
+    duration: 0.75,
+    ease: "linear",
+    times: [0, 1],
+  };
+  const transitionLineMobile = {
+    duration: 0.75,
+    ease: "easeIn",
+    times: [0, 1],
+  };
+  const initialTitle = { y: "84px" };
+
+  function display(place) {
+    return position === place ? "block" : "none";
+  }
+
+  function animateMessage(place) {
+    if (dimensions.width > breakpointMobile) {
+      return position === place
+        ? { x: [-117, 0, 0, 310, 310], opacity: [0, 1, 1, 0, 0] }
+        : "";
+    }
+
+    return position === place
+      ? {
+          x: [-117, 0, 0, 0, 0],
+          y: [0, 0, 0, 75, 75],
+          opacity: [0, 1, 1, 1, 1],
+        }
+      : "";
+  }
+
+  function animateTitle(place) {
+    return position === place ? { y: 0 } : "";
+  }
 
-  function getClassName(place) {
-    if (place === position) {
-      return "line-klein-blue";
+  function animateLine(place) {
+    if (dimensions.width > breakpointMobile) {
+      return position === place ? { width: ["calc(15px)", "calc(117px)"] } : "";
     }
 
-    return "line-baby-blue";
+    return position === place ? { width: ["calc(10%)", "calc(100%)"] } : "";
   }
 
-  function reset(value) {
-    setSeconds(value);
-    setPosition(value);
+  function reset(place) {
+    setSeconds(place);
+    setPosition(place);
   }
 
   useEffect(() => {
     let interval = null;
 
     interval = setInterval(() => {
-      setSeconds(secs => secs + 1);
-      setPosition((seconds + 1) % 3);
-    }, 2500);
+      const newSeconds = seconds + 1;
+
+      setSeconds(newSeconds);
+      setPosition(newSeconds % 3);
+    }, 4000);
 
     return () => clearInterval(interval);
-  }, [seconds]);
-
-  return (
-    <div styleName="copy-section">
-      <div styleName="lines">
-        <button
-          type="button"
-          aria-label="clickable info 1"
-          styleName={getClassName(0)}
-          style={{ "--top": "0" }}
-          onClick={() => reset(0)}
-        />
-        <button
-          type="button"
-          aria-label="clickable info 2"
-          styleName={getClassName(1)}
-          style={{ "--top": "1" }}
-          onClick={() => reset(1)}
-        />
-        <button
-          type="button"
-          aria-label="clickable info 3"
-          styleName={getClassName(2)}
-          style={{ "--top": "2" }}
-          onClick={() => reset(2)}
-        />
-      </div>
-      <div styleName="message">
-        <Typography color="oxford-blue">
-          {textPlaceholder[position].message}
-        </Typography>
-        <Typography color="klein-blue" weight="bold" variant="h2">
-          {textPlaceholder[position].title}
-        </Typography>
-      </div>
-    </div>
-  );
+  }, [seconds, position]);
+
+  function renderLine(transition, place) {
+    return (
```

Comment:
Now I think you can put `animateLine` logic here

Reply:
You mean put the `function animateLine()` inside?

Reply:
Only the logic (if....). That logic is relative to line, isn't it?

Reply:
Yes! I thought you meant the whole function, but yeah, makes sense only the ìf()`part!